src/univalue/univalue_read.cpp

```diff
@@ -188,7 +188,7 @@ enum jtokentype getJsonToken(string& tokenVal, unsigned int& consumed,
                 case 't':  valStr += "\t"; break;
 
                 case 'u': {
-                    char buf[4] = {0,0,0,0};
+                    char buf[3];
```

Comment:
This absolutely needs review, but the changes look very sane to me. The buffer size is decreased from 4 to 3 because the last slot was only there to hold a terminating NUL character. This is no longer necessary because the new code counts characters. If this introduces an overflow bug, it was already there.()

I wonder one thing though: could we `push_back` the characters into `valStr` directly instead of the intermediate `buf`? This would avoid any overflow risk.